core/templates/dev/head/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,112 @@
+        opportunityDict.topic_name, opportunityDict.question_count);
+    }
+  }
+
+  _fetchOpportunities(
+      opportunityType: OpportunityType,
+      params, successCallback, errorCallback
+  ): void {
+    this.http.get(this.urlInterpolationService.interpolateUrl(
+      this.urlTemplate, { opportunityType }
+    ), { params }).toPromise().then((data: any) => {
+      const opportunities = [];
+      for (const index in data.opportunities) {
+        opportunities.push(this._getOpportunityFromDict(
+          opportunityType, data.opportunities[index]));
       }
-    };
-    var _fetchOpportunities = function(
-        opportunityType, params, successCallback) {
-      return $http.get(
-        UrlInterpolationService.interpolateUrl(
-          urlTemplate, {opportunityType: opportunityType}
-        ), {
-          params: params
-        }).then(function(response) {
-        var data = response.data;
-        var opportunities = [];
-        for (var index in data.opportunities) {
-          opportunities.push(_getOpportunityFromDict(
-            opportunityType, data.opportunities[index]));
-        }
+      if (successCallback) {
         successCallback(opportunities, data.next_cursor, data.more);
-      });
-    };
-    return {
-      fetchSkillOpportunities: function(cursor, successCallback) {
-        var params = {
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_SKILL, params, successCallback);
-      },
-      fetchTranslationOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_TRANSLATION, params, successCallback);
-      },
-      fetchVoiceoverOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_VOICEOVER, params, successCallback);
       }
+    }, (error) => {
+      if (errorCallback) {
+        errorCallback(error);
+      }
+    });
+  }
+
+  fetchSkillOpportunities(
+      cursor,
+      successCallback,
+      errorCallback?
+  ): void {
+    const params = {
+      cursor: cursor
     };
-  }]);
+    return this._fetchOpportunities(
+      constants.OPPORTUNITY_TYPE_SKILL, params, successCallback, errorCallback);
+  }
+
+  fetchTranslationOpportunities(
+      languageCode: String,
+      cursor,
+      successCallback,
+      errorCallback?
+  ): void {
+    const params = {
+      language_code: languageCode,
+      cursor: cursor
+    };
+    return this._fetchOpportunities(
+      constants.OPPORTUNITY_TYPE_TRANSLATION,
+      params, successCallback, errorCallback);
+  }
+
+  fetchVoiceoverOpportunities(
+      languageCode: String,
+      cursor,
+      successCallback,
```

Comment:
Add types for these.

Reply:
If you are unsure about the type and unable to find that out too, add the type as any and add a todo
```
// TODO(#7165): Replace any with exact type.
```
As a reference, take a look at **language-util.service.ts**

Reply:
Done